src/main/java/solution/BacktrackingSolution.java

```diff
@@ -0,0 +1,42 @@
+package solution;
+
+import java.awt.*;
+import java.util.List;
+
+class BacktrackingSolution implements Solution {
+
+    private BacktrackingSolution() {
+    }
+
+    static SudokuBoard solve(SudokuBoard board) {
+        if (board == null) {
+            throw new NullPointerException("Board cannot be null.");
+        }
+        System.out.println("Initial board.");
```

Comment:
Normally, you would use logging, preferably `slf4j` as an API and `logback` as its implementation. Although it's fine to use `System.out.println` for a project like this one. I will leave it up to you if you want to play with logging here.